wasm/validator/src/lib.rs

```diff
@@ -1,422 +1,135 @@
 //! API for *Runtime Validators*.
-
 #![no_std]
 
 extern crate alloc;
-
-use alloc::string::String;
-
-use iroha_wasm::data_model::{prelude::*, validator::Verdict};
-pub use iroha_wasm::{self, data_model, ExecuteOnHost};
-
-pub mod prelude {
-    //! Contains useful re-exports
-
-    pub use iroha_validator_derive::{entrypoint, Token, ValidateGrantRevoke};
-    pub use iroha_wasm::{
-        data_model::{prelude::*, validator::Verdict},
-        prelude::*,
-        Context,
+// NOTE: Enables us to implement derive macros
+// that refer to the this crate for this crate
+#[cfg(feature = "default-validator")]
+extern crate self as iroha_validator;
+
+#[cfg(feature = "default-validator")]
+pub use default::DefaultValidator;
+use iroha_wasm::data_model::{
+    validator::{DenialReason, Verdict},
+    visit::Visit,
+};
+pub use iroha_wasm::{self, data_model};
+
+#[cfg(feature = "default-validator")]
+pub mod default;
+pub mod permission;
+
+/// Shortcut for `return Ok(())`.
+#[macro_export]
+macro_rules! pass {
+    ($validator:ident) => {
+        #[cfg(debug_assertions)]
+        if let Err(_error) = $validator.verdict() {
+            panic!("Validator already denied");
+        }
+
+        return
     };
```

Comment:
So here we use `panic` (see next comment

Reply:
yeah, should be `unreachable` as well

Reply:
not a big deal, I'm going to merge now and fix in another PR